### PR TITLE
Add ssd-sharded-rocksdb storage engine type.

### PR DESCRIPTION
**Pull request by yao-xiao-github**
# Description

*Please include a summary of the change and which issue is addressed. If this change resolves an issue, please include the issue number in the description.*

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation
- Other

## Discussion

*Are there any design details that you would like to discuss further?*

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

## Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

## Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

**Review by johscheuer — COMMENTED** [review]
The changes LGTM, could you run  the following commands:

- `make clean all`, this will ensure that the required code is generated an the yaml file is upated
- `make fmt lint`, this will format the go code and ensure there are no linting issues

Once those things are done we can merge this PR 👍

File: config/rbac/role.yaml-e [hunk 1]
```diff
@@ -0,0 +1,109 @@
+
```
Comment (johscheuer) [inline config/rbac/role.yaml-e:1]: I think this file must be removed and was added by accident?

**Review by johscheuer — COMMENTED** [review]
Once the `config/rbac/role.yaml-e` file is removed we can merged this PR. Thanks!

File: config/rbac/role.yaml-e (hunk 1 above)
Comment (yao-xiao-github) [reply config/rbac/role.yaml-e]: This file should not be tracked. I somehow added it.

**Comment by yao-xiao-github** [comment]
Friendly ping.

**Review by johscheuer — APPROVED** [review]

**Pull request merged**